client/app/components/empty-state/EmptyState.jsx

```diff
@@ -17,9 +19,9 @@ export function Step({ show, completed, text, url, urlTarget, urlText, onClick }
 
   return (
     <li className={classNames({ done: completed })}>
-      <Link href={url} onClick={onClick} target={urlTarget}>
+      <Button type="link" href={url} onClick={onClick} target={urlTarget}>
```

Comment:
`<Link>` _should_ throw a type error if it get's a click handler. This was sort of a workaround. I'm trying to think in a more general solution to this.

Reply:
`<Link>` has some logic that allows to modify URL. Using `<Button>` here will break those links in Databricks internal fork

Reply:
I do not agree that link should throw a type  error if it gets a click handler. 

It's valid to have a link perform one behavior when you click on it, and another if you middle click / open in a new tab (for example, if you want to open a modal, you don't want to refresh the page when you left click, but you do want a new page with the modal if you open a link to it in a new tab).

Reply:
The problem in this case is for pieces of code like this: 
<img width="526" alt="image" src="https://user-images.githubusercontent.com/44540213/113869930-ab954e00-9787-11eb-8a77-ec69c5c6ab8f.png">

In practice, this component is sometimes a link and sometimes a button, which hurts its accessibility.

Reply:
When `Step` component has a link - it can have either `urlText` + `onClick` or `urlText` + `url` + `urlTarget`. You can use this to decide what to render - link or button. Also, you could even update TS types to indicate this:

```ts
type StepCommonProps = { ... };

type StepLinkProps = {
  urlText: string;
  url: string;
  urlTarget: string;
}

type StepButtonProps = {
  urlText: string;
  onClick: ...;
}

type StepProps = StepCommonProps & (StepLinkProps | StepButtonProps);
```

In this case when you'll try to render `<Step>` component with `onClick` - TS will not allow you to use `url` and `urlTarget`, and if you'll add `url` and/or `urlTarget` - it will forbid `onClick`

Reply:
Update: see #5435